docs/en/integrations/language-clients/rust.md

```diff
@@ -0,0 +1,517 @@
+---
+sidebar_label: Rust
+sidebar_position: 4
+keywords: [clickhouse, rs, rust, http, client, connect, integrate]
+slug: /en/integrations/rust
+description: The official Rust client for connecting to ClickHouse.
+---
+
+# ClickHouse Rust Client
+
+The official Rust client for connecting to ClickHouse, originally developed by [Paul Loyd](https://github.com/loyd). The client source code is available in the [GitHub repository](https://github.com/ClickHouse/clickhouse-rs).
+
+## Overview
+
+* Uses `serde` for encoding/decoding rows.
+* Supports `serde` attributes: `skip_serializing`, `skip_deserializing`, `rename`.
+* Uses [`RowBinary`](https://clickhouse.com/docs/en/interfaces/formats#rowbinary) format over the HTTP transport.
+    * There are plans to switch to [`Native`](https://clickhouse.com/docs/en/interfaces/formats#native) over TCP.
+* Supports TLS (via `native-tls` and `rustls-tls` features).
+* Supports compression and decompression (LZ4).
+* Provides APIs for selecting or inserting data, executing DDLs, and client-side batching.
+* Provides convenient mocks for unit testing.
+
+## Installation
+
+To use the crate, add the following to your `Cargo.toml`:
+
+```toml
+[dependencies]
+clickhouse = "0.12.2"
+
+[dev-dependencies]
+clickhouse = { version = "0.12.2", features = ["test-util"] }
+```
+
+See also: [crates.io page](https://crates.io/crates/clickhouse).
+
+## Cargo features
+
+* `lz4` (enabled by default) — enables `Compression::Lz4` and `Compression::Lz4Hc(_)` variants. If enabled, `Compression::Lz4` is used by default for all queries except for `WATCH`.
+* `native-tls` — supports urls with the `HTTPS` schema via `hyper-tls`, which links against OpenSSL.
+* `rustls-tls` — supports urls with the `HTTPS` schema via `hyper-rustls`, which does not link against OpenSSL.
+* `inserter` — enables `client.inserter()`.
+* `test-util` — adds mocks. See [the example](https://github.com/ClickHouse/clickhouse-rs/tree/main/examples/mock.rs). Use it only in `dev-dependencies`.
+* `watch` — enables `client.watch` functionality. See the corresponding section for details.
+* `uuid` — adds `serde::uuid` to work with [uuid](https://docs.rs/uuid) crate.
+* `time` — adds `serde::time` to work with [time](https://docs.rs/time) crate.
+
+:::important
+When connecting to ClickHouse via an `HTTPS` url, either the `native-tls` or `rustls-tls` feature should be enabled.
+If both are enabled, the `rustls-tls` feature will take precedence.
+:::
+
+## ClickHouse versions compatibility
+
+The client is compatible with the LTS or newer versions of ClickHouse, as well as ClickHouse Cloud.
+
+ClickHouse server older than v22.6 handles RowBinary [incorrectly in some rare cases](https://github.com/ClickHouse/ClickHouse/issues/37420). 
```

Comment:
Let's mention the version we test against: It should be a maintained version. As of today, it's `24.3`, and `24.6+`

Reply:
It will be added later in a follow-up when we change our test matrix.